test/no-array-instanceof.js

```diff
@@ -0,0 +1,69 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-array-instanceof';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'no-array-instanceof',
+	message: 'Use `Array.isArray()` instead of `instanceof Array`'
+}];
+
+ruleTester.run('avoid-array-instanceof', rule, {
+	valid: [
+		'Array.isArray(arr)',
+		'Array.isArray(arr) === true',
+		'Array.isArray([])',
+		'Array.isArray([]) === true',
+		'Array.isArray([1,2,3])',
+		'Array.isArray([1,2,3]) === true',
+		'if(Array.isArray(arr)){}',
+		'if(Array.isArray([1,2,3])){}'
```

Comment:
In the sense of testing your rule, these cases do not test the rule's boundaries, as you never test the presence of `Array.isArray`. You should not remove them, but you should add tests containing `instanceof` comparisons and the sorts. Here are a few examples

// Valid
```js
a instanceof array;
a instanceof 'array';
a instanceof ArrayA;
a instanceof Object;
```

// Invalid
```js
a.b instanceof Array;
foo() instanceof Array;
foo`bar${value}` instanceof Array;
```